### PR TITLE
[1/n] upgrade react-native 0.75

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

upgrade react-native to 0.75 on main, the idea is that it should be backward compatible with SDK 51 

Closes ENG-12562

# How

- update package versions
  - `react-native 0.74.3 -> 0.75.0-rc.5` 
  - `react 18.2.0 -> 19.0.0-rc-fb9a90fa48-20240614`
  - `react-dom 18.2.0 -> 19.0.0-rc-fb9a90fa48-20240614`  
  - `react-test-renderer 18.2.0 -> 19.0.0-rc-fb9a90fa48-20240614`  
- upgrade project templates based on [upgrade-helper](https://react-native-community.github.io/upgrade-helper/?from=0.74.2&to=0.75.0-rc.2)
- Set CXX language standard to c++20 on expo-av
- Reverted https://github.com/expo/expo/pull/30239 because Kotlin getters were reverted in RC5

- [bare-expo][fabric-tester] Temporarily disabled DevClient (reenabled here -> https://github.com/expo/expo/pull/30545)
- [cli ] Add `--config-cmd` option to `export:embed` (for additional info check https://github.com/facebook/react-native/pull/45560)
- [expo-go]react-native upgrade -> https://github.com/expo/expo/pull/30519

#### Libraries that need to be patched for now:
- react-native-reanimated
- react-native-pager-view
- react-native-safe-area-context 
- react-native-screens 
- react-native

# Test Plan

- FabricTester
   -  Crashing on Android
- BareExpo

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/bare-expo/MainNavigator.tsx [hunk 1]
```diff
@@ -50,6 +49,8 @@ const Search = optionalRequire(() =>
   require('native-component-list/src/screens/SearchScreen')
 ) as any;
 
+const DevMenu = optionalRequire(() => require('expo-dev-menu')) as any;
```
Comment (gabrieldonadel) [inline apps/bare-expo/MainNavigator.tsx:52]: Temporary workaround while dev-client is disabled

File: apps/fabric-tester/App.tsx [hunk 2]
```diff
@@ -257,7 +257,9 @@ const styles = StyleSheet.create({
     fontSize: 16,
     fontWeight: 'bold',
     textAlign: 'center',
-    color: PlatformColor('labelColor'),
+    ...Platform.select({
+      ios: { color: PlatformColor('labelColor') },
+    }),
```
Comment (gabrieldonadel) [inline apps/fabric-tester/App.tsx:262]: calling PlatformColor `labelColor` was causing a hard crash on android

File: packages/expo-updates/e2e/setup/project.ts [hunk 3]
```diff
@@ -191,6 +191,13 @@ async function copyCommonFixturesToProject(
   // copy .prettierrc
   await fs.copyFile(path.resolve(repoRoot, '.prettierrc'), path.join(projectRoot, '.prettierrc'));
 
+  // Copy react-native patch
+  await fs.mkdir(path.join(projectRoot, 'patches'));
+  await fs.copyFile(
+    path.resolve(repoRoot, 'patches', 'react-native+0.75.0-rc.5.patch'),
+    path.join(projectRoot, 'patches', 'react-native+0.75.0-rc.5.patch')
+  );
```
Comment (gabrieldonadel) [inline packages/expo-updates/e2e/setup/project.ts:199]: Necessary for e2e tests while https://github.com/facebook/react-native/pull/45560 is not merged

**Review by wschurman — APPROVED** [review]
Changes to expo-updates and updates e2e test setup look good. Didn't review the rest of the PR, but can if you'd like me as a primary reviewer.

**Review by tsapeta — APPROVED** [review]

**Review by alanjhughes — APPROVED** [review]

**Pull request merged**